hpx/runtime/threads/policies/shared_priority_queue_scheduler.hpp

```diff
@@ -323,7 +323,7 @@ namespace hpx { namespace threads { namespace policies {
                     {
                         debug::set(msg, "HINT_NONE  ");
                         // Create thread on this worker thread if possible
-                        if (local_num < 0)
+                        if (local_num == UINT64_MAX)
```

Comment:
Good catch! We usually do something like this:
```suggestion
                        if (local_num == std::size_t(-1))
```
especially since `size_t` isn't guaranteed to be 64 bits long everywhere.

Reply:
Thanks!